enhancements/machine-config/mco-cgroupsv2-support.md

```diff
@@ -121,19 +122,19 @@ type NodeSpec struct {
 
 ### Operational Aspects of API Extensions
 
-Once the previous API is defined, the MCO will read the configured object and
-set the appropriate kernel options (on bootstrap). The MCO will report an error
+- MCO reads the configured object and
+sets the appropriate kernel options (on bootstrap).
+- The MCO will report an error
 if a user tries to modify/add cgroup related kargs within a MachineConfig.
-
-
-The following kernel command line arguments would be set when `CgroupMode_v2` is enabled:
+- An admission hook is introduced to allow if a cluster is transitioning from "cgroup v1" -> "cgroup v2" and deny the other way of "cgroup v2" -> "cgroup v1"
+- MCO also reports error if a user tries to set the `CgroupMode` to `CgroupMode_V1` and sets the cluster operator's status condition of `Upgradeble=False`
```

Comment:
Could you specify how this error will show up for the user? Maybe we should reject at the API level (see below)

Reply:
yes, the idea is to add an admission webhook to prevent setting of `CgroupMode_V1` as mentioned in the Goals section.
MCO logs and returns the error from reconcile loop as if the `CgroupMode_V1` is similar to any unknown value.
Ref: https://github.com/openshift/machine-config-operator/blob/0f9a11b4b5c89912eea695d5188be48c276ea94a/pkg/controller/kubelet-config/helpers.go#L163